src/hooks/use-snapshots.tsx

```diff
@@ -0,0 +1,376 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { useI18n } from '@wordpress/react-i18n';
+import React, {
+	createContext,
+	useContext,
+	useState,
+	useCallback,
+	useEffect,
+	ReactNode,
+	useMemo,
+	useLayoutEffect,
+} from 'react';
+import { LIMIT_OF_ZIP_SITES_PER_USER } from '../constants';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { useAuth } from './use-auth';
+import { useOffline } from './use-offline';
+
+interface SnapshotContextType {
+	snapshots: Snapshot[];
+	allSnapshots: Pick< Snapshot, 'atomicSiteId' >[] | null;
+	addSnapshot: ( snapshot: Snapshot ) => void;
+	updateSnapshot: ( snapshot: Partial< Snapshot > ) => void;
+	removeSnapshot: ( snapshot: Pick< Snapshot, 'atomicSiteId' > ) => void;
+	fetchAllSnapshots: () => Promise< Pick< Snapshot, 'atomicSiteId' >[] | null >;
+	deleteSnapshot: (
+		snapshot: Pick< Snapshot, 'atomicSiteId' >,
+		displayAlert?: boolean
+	) => Promise< void >;
+	deleteAllSnapshots: ( snapshots: Pick< Snapshot, 'atomicSiteId' >[] ) => Promise< void >;
+	clearFloatingSnapshots: ( allSnapshots: Pick< Snapshot, 'atomicSiteId' >[] ) => void;
+	isLoading: boolean;
+	loadingDeletingAllSnapshots: boolean;
+	loadingServerSnapshots: boolean;
+	activeSnapshotCount: number;
+	snapshotQuota: number;
+	fetchSnapshotUsage: () => Promise< { site_limit: number; site_count: number } | undefined >;
+	isLoadingSnapshotUsage: boolean;
+	initiated: boolean;
+}
+
+export interface SnapshotStatusResponse {
+	is_deleted: string;
+	domain_name: string;
+	atomic_site_id: string;
+	status: '1' | '2';
+}
```

Comment:
What do you think about using an enum for the status?

```suggestion
enum SnapshotStatus {
	Deleted = '1',
	Active = '2',
}

export interface SnapshotStatusResponse {
	is_deleted: string;
	domain_name: string;
	atomic_site_id: string;
	status: SnapshotStatus;
}
```